modules/dxp/apps/digital-sales-room/headless/headless-digital-sales-room-impl/src/main/java/com/liferay/headless/digital/sales/room/internal/resource/v1_0/DigitalSalesRoomResourceImpl.java

```diff
@@ -37,8 +83,211 @@ public Page<DigitalSalesRoom> getDigitalSalesRoomsPage(
+
+		_objectEntryLocalService.updateObjectEntry(objectEntry);
+
+		return _toDigitalSalesRoom(group);
+	}
+
+	private Group _addGroup(
+			String externalReferenceCode, String description,
+			String friendlyURLPath, String siteTemplateKey, String name)
+		throws Exception {
+
+		SiteInitializer siteInitializer =
+			_siteInitializerRegistry.getSiteInitializer(siteTemplateKey);
+
+		if (siteInitializer == null) {
+			throw new IllegalArgumentException(
+				"No site initializer was found for site template key " +
+					siteTemplateKey);
+		}
+
+		_initThemeDisplay();
+
+		try (AutoCloseable autoCloseable =
+				_layoutServiceContextHelper.getServiceContextAutoCloseable(
+					contextCompany, contextUser)) {
+
+			Group group = _groupService.addGroup(
+				externalReferenceCode, GroupConstants.DEFAULT_PARENT_GROUP_ID,
+				GroupConstants.DEFAULT_LIVE_GROUP_ID,
+				HashMapBuilder.put(
+					LocaleUtil.getDefault(), name
+				).build(),
+				HashMapBuilder.put(
+					LocaleUtil.getDefault(), description
+				).build(),
+				GroupConstants.TYPE_SITE_RESTRICTED, null, true,
+				GroupConstants.DEFAULT_MEMBERSHIP_RESTRICTION, friendlyURLPath,
+				true, false, true, _getServiceContext());
+
+			LiveUsers.joinGroup(
+				contextCompany.getCompanyId(), group.getGroupId(),
+				contextUser.getUserId());
+
+			siteInitializer.initialize(group.getGroupId());
+
+			return group;
+		}
+		catch (Exception exception) {
+
+			// LPS-169057
+
+			PermissionCacheUtil.clearCache(contextUser.getUserId());
+
+			throw exception;
+		}
+		finally {
+			ServiceContextThreadLocal.popServiceContext();
+		}
+	}
+
+	private ServiceContext _getServiceContext() throws PortalException {
+		ServiceContext serviceContext = null;
+
+		if (contextHttpServletRequest != null) {
+			serviceContext = ServiceContextFactory.getInstance(
+				contextHttpServletRequest);
+		}
+		else {
+			serviceContext = new ServiceContext();
+
+			serviceContext.setCompanyId(contextCompany.getCompanyId());
+			serviceContext.setRequest(contextHttpServletRequest);
+			serviceContext.setUserId(contextUser.getUserId());
+		}
+
+		ServiceContextThreadLocal.pushServiceContext(serviceContext);
+
+		return serviceContext;
 	}
 
+	private void _initThemeDisplay() throws Exception {
+		if (contextHttpServletRequest == null) {
+			return;
+		}
+
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)contextHttpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		if (themeDisplay != null) {
+			return;
+		}
+
+		ServicePreAction servicePreAction = new ServicePreAction();
+
+		servicePreAction.servicePre(
+			contextHttpServletRequest, contextHttpServletResponse, false);
+
+		ThemeServicePreAction themeServicePreAction =
+			new ThemeServicePreAction();
+
+		themeServicePreAction.run(
+			contextHttpServletRequest, contextHttpServletResponse);
+
+		themeDisplay = (ThemeDisplay)contextHttpServletRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		themeDisplay.setResponse(new DummyHttpServletResponse());
```

Comment:
Why a dummy response?

Reply:
copied but I think it's because it's called by a "service" so the response for the themeDisplay is not useful